src/utils/hooks/useTextKeys.tsx

```diff
@@ -0,0 +1,36 @@
+import { TranslationsContext } from '@hedviginsurance/textkeyfy'
+import * as React from 'react'
+
+const placeholderRegex = new RegExp('({[a-zA-Z0-9_]+})', 'g')
+const placeholderKeyRegex = new RegExp('([a-zA-Z0-9_]+)', 'g')
+
+String.prototype.fill = function(replacements) {
```

Comment:
This is awesome!!

__HOWEVER__ i think we shouldn't "bloat" the global string object (`fill` is quite generic and it could easily get overridden by something else, for example). I think we might as well do this for each text key we return or? I think something like 

```ts

function fill(replacements: Record<string, any>) { ... }

// ...

 return new Proxy(context.textKeys, {
      get: (textKeys, key: string) => {
        const actualKey = textKeys[key] || key
        actualKey.fill = fill
        return actualKey
      },
    })
```

Reply:
I think TypeScript might hate us if we do this (string.fill is not a function or something like that)

I agree that not filling up the string prototype would be nice. We could perhaps return a function from the proxy so that you can do

```ts

const textKeys = useTextKeys()

textKeys.SOME_KEY_WITH_REPLACEMENTS({REPLACEMENT: 'VALUE'})
textKeys.SOME_REGULAR_KEY()
```

Reply:
I tried to wing an implementation of this but it didnt turn out good enough so idk

Reply:
Here is that implementation for reference @sampettersson @palmenhq 

```ts
import { Replacements, TranslationsContext } from '@hedviginsurance/textkeyfy'
import * as React from 'react'
const placeholderRegex = new RegExp('({[a-zA-Z0-9_]+})', 'g')
const placeholderKeyRegex = new RegExp('([a-zA-Z0-9_]+)', 'g')
export const useTextKeys = () => {
  const context = React.useContext(TranslationsContext)
  return React.useMemo(
    () =>
      new Proxy<{ [key: string]: () => string }>(
        {},
        {
          get: (target, key: string) => (
            replacements?: Replacements,
          )string => {
            const possiblyUnresolvedKey = context.textKeys[key] || key
            if (!replacements) {
              return possiblyUnresolvedKey
            }
            const matches = possiblyUnresolvedKey
              .split(placeholderRegex)
              .filter((value) => value)
            return matches
              .map((placeholder, _) => {
                if (!placeholderKeyRegex.test(placeholder)) {
                  return placeholder
                }
                const replacementKey = placeholder.match(
                  placeholderKeyRegex,
                )![0]
                const value = replacements[replacementKey]
                if (value) {
                  return value.toString()
                }
                return placeholder
              })
              .reduce((acc, curr) => acc.concat(curr), '')
          },
        },
      ),
    [context.textKeys],
  )
}
```

Reply:
I tried this again, but typescript is puking on me... I will change the function name to `withReplacements` as @sampettersson suggested, making the name more unique and less likely to be overwritten

Reply:
TypeScript actually puked on me with the current implementation: `Property 'fill' does not exist on type 'String'.ts(2339)`

Reply:
Yep, same thing for me. Let's go with `withReplacements` for now, and if someone solves this, they are more than welcome to update the hook logic 😄

Reply:
Can't we make a composite return type?

```
 new Proxy<{ [key: string]: (string & { fill: (..) => string }) }>(...)
```

or so?

Reply:
If typescript accepts this it would be nice!

Reply:
We messed with it a little bit and came back to the version with keys being functions that take an optional parameter.
Slightly less fun to use, but we avoid messing with the string prototype chain and typescript actually liked it this time which is nice